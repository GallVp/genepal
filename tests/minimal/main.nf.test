nextflow_pipeline {

    name "Test corresponding to 'test' profile"
    script "main.nf"

    test("profile - test") {

        when {
            params {
                input               = "$baseDir/tests/minimal/assemblysheet.csv"
                protein_evidence    = "https://raw.githubusercontent.com/Gaius-Augustus/BRAKER/f58479fe5bb13a9e51c3ca09cb9e137cab3b8471/example/proteins.fa"
                braker_extra_args   = "--gm_max_intergenic 10000 --skipOptimize"
                busco_skip          = true
                outdir              = "$outputDir"
            }
        }

        then {
            def stable_path = getAllFilesFromDir(
                params.outdir,
                false,
                [
                    'pipeline_info/*.{html,json,txt,yml}',
                    'multiqc_report.html'
                ],
                null,
                ['**']
            )

            def stable_name = getAllFilesFromDir(
                params.outdir,
                true,
                [
                    'pipeline_info/*.{html,json,txt,yml}'
                ],
                null,
                ['**']
            )

            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    [
                        'successful tasks': workflow.trace.succeeded().size(),
                        'versions': removeNextflowVersion("$outputDir/pipeline_info/genepal_software_mqc_versions.yml"),
                        'stable paths': stable_path,
                        'stable names': getRelativePath(stable_name, outputDir),
                    ]
                ).match() }
            )
        }

    }

}
