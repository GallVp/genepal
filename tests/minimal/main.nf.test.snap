{
    "profile - test": {
        "content": [
            {
                "successful tasks": 21,
                "versions": {
                    "AGAT_CONVERTSPGFF2GTF": {
                        "agat": "v1.4.0"
                    },
                    "AGAT_CONVERTSPGXF2GXF": {
                        "agat": "v1.4.0"
                    },
                    "AGAT_SPADDINTRONS": {
                        "agat": "v1.4.0"
                    },
                    "AGAT_SPEXTRACTSEQUENCES": {
                        "agat": "v1.4.0"
                    },
                    "BRAKER3": {
                        "braker3": "3.0.8",
                        "augustus": "3.5.0",
                        "genemark-etp": 1.02,
                        "prothint": "2.6.0"
                    },
                    "CAT_PROTEIN_FASTAS": {
                        "pigz": "2.3.4"
                    },
                    "EXTRACT_CDNA": {
                        "gffread": "0.12.7"
                    },
                    "EXTRACT_CDS": {
                        "gffread": "0.12.7"
                    },
                    "EXTRACT_PROTEINS": {
                        "gffread": "0.12.7"
                    },
                    "FASTAVALIDATOR": {
                        "py_fasta_validator": 0.6
                    },
                    "FILTER_BY_ORF_SIZE": {
                        "agat": "v1.4.3"
                    },
                    "FINAL_GFF_CHECK": {
                        "genometools": "1.6.5"
                    },
                    "GFF2FASTA_FOR_EGGNOGMAPPER": {
                        "gffread": "0.12.7"
                    },
                    "GT_GFF3": {
                        "genometools": "1.6.5"
                    },
                    "SAVE_MARKED_GFF3": {
                        "pigz": "2.3.4"
                    },
                    "SEQKIT_RMDUP": {
                        "seqkit": "v2.8.0"
                    },
                    "STAR_GENOMEGENERATE": {
                        "star": "2.7.10a",
                        "samtools": 1.18,
                        "gawk": "5.1.0"
                    },
                    "TSEBRA": {
                        "tsebra": "1.1.2.5"
                    },
                    "Workflow": {
                        "plant-food-research-open/genepal": "v0.6.1dev"
                    }
                },
                "stable paths": [
                    "a_thaliana.cdna.fasta:md5,f44a6acdc245b85a935ef9ed3fc62967",
                    "a_thaliana.cds.fasta:md5,4fd5a251a3912b452844e1f7332f514d",
                    "a_thaliana.gt.gff3:md5,00b2ee481731e9ada62575ebe5eff1e4",
                    "a_thaliana.pep.fasta:md5,38ae660abc78eb57d467bb48b5170f19",
                    "a_thaliana.gff3:md5,bc520744a70fd9bf1b4bd4271bc83fc4",
                    "summary_stats.json:md5,007ba5cf2b7a2fd395a27d9458ca2d2e"
                ],
                "stable names": [
                    "annotations",
                    "annotations/a_thaliana",
                    "annotations/a_thaliana/a_thaliana.cdna.fasta",
                    "annotations/a_thaliana/a_thaliana.cds.fasta",
                    "annotations/a_thaliana/a_thaliana.gt.gff3",
                    "annotations/a_thaliana/a_thaliana.pep.fasta",
                    "etc",
                    "etc/splicing_marked",
                    "etc/splicing_marked/a_thaliana.gff3",
                    "genepal_data",
                    "genepal_data/summary_stats.json",
                    "genepal_report.html",
                    "multiqc_report.html",
                    "pipeline_info"
                ],
                "summary_stats": {
                    "stats": [
                        {
                            "ID": "a_thaliana",
                            "Genes": 252,
                            "mRNA": 265,
                            "CDS": 1340,
                            "Exons": 1340,
                            "Intron": 1075,
                            "Non canon splice sites": 18
                        }
                    ]
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-08T09:54:41.452661"
    }
}